dart/biomechanics/DynamicsFitter.cpp

```diff
@@ -13134,108 +13144,196 @@ void DynamicsFitter::recomputeGRFs(
-                = closestFoot;
-
-            Eigen::Vector3s force = forcePlates[i].forces[assignT];
-            Eigen::Vector3s cop = forcePlates[i].centersOfPressure[assignT];
-            Eigen::Vector3s moments = forcePlates[i].moments[assignT];
-
-            Eigen::Vector6s wrench = Eigen::Vector6s::Zero();
-            wrench.head<3>() = moments;
-            wrench.tail<3>() = force;
-            Eigen::Isometry3s wrenchT = Eigen::Isometry3s::Identity();
-            wrenchT.translation() = cop;
-            Eigen::Vector6s worldWrench = math::dAdInvT(wrenchT, wrench);
-
-            // If multiple force plates assign to the same foot, sum up the
-            // forces
-            init->grfTrials[trial].block<6, 1>(closestFoot * 6, assignT)
-                += worldWrench;
+            assignedToFeet.push_back(closestFoot);
           }
         }
-        lastStartedTrack = -1;
-        sumSquaredDistances.setZero();
-        continue;
       }
 
-      if (lastStartedTrack == -1)
-        lastStartedTrack = t;
-      for (int b = 0; b < init->grfBodyNodes.size(); b++)
+      if (assignedToFeet.size() == 0)
       {
-        sumSquaredDistances(b) += (footLocations[t][b] - cop).squaredNorm();
+        // This force plate is never assigned to a foot, so we can't do anything
+        std::cout << "NOTE: Assigning force plate " << i
+                  << " to no feet during trial " << trial
+                  << ", because there was never sufficient measured force on "
+                     "the plate"
+                  << std::endl;
+      }
+      else if (assignedToFeet.size() == 1)
+      {
+        std::cout << "Assigning force plate " << i << " to foot "
+                  << assignedToFeet[0] << " on trial " << trial << std::endl;
+        // This force plate is only ever assigned to a single foot, so we can
+        // assign all timesteps to that foot
+        for (int t = 0; t < poses.cols(); t++)
+        {
+          // This is a map of [trial][forcePlate][timestep]
+          init->forcePlatesAssignedToContactBody[trial][i][t]
+              = assignedToFeet[0];
+        }
+      }
+      else
+      {
+        std::cout << "NOTE: Assigning force plate " << i
+                  << " to multiple feet during trial " << trial << std::endl;
+        // This force plate is assigned to multiple feet over time. This can
+        // often happen with foot-crossing. We won't issue a warning, but we can
+        // go through and try to fill in the gaps between feet assignments that
+        // are the same.
+        int lastContactIndex = -1;
+        for (int t = 0; t < poses.cols(); t++)
+        {
+          int closestFoot = init->forcePlatesAssignedToContactBody[trial][i][t];
+          if (closestFoot != -1)
+          {
+            // If we find a gap (filled with -1's) where the contact foot on
+            // both ends is the same, then we can fill it in
```

Comment:
Will this gap filling effectively interpolate COPs?

Reply:
This will just copy everything from the input data (tiny force noise, COPs, etc) from the force plate onto this foot over the whole gap,  rather than setting it all to 0s like we used to do.